collector-state/src/main/java/com/iluwatar/collectorstate/App.java

```diff
@@ -0,0 +1,39 @@
+package com.iluwatar.collectorstate;
+
+import lombok.extern.slf4j.Slf4j;
+
+
+/**
+ * This is a brief structural introduction to Collector-State pattern using state machine
+ * with self-defined message and states.

```

Comment:
Describe the pattern briefly and then explain how the pattern is implemented in this example